lib/src/components/NxDropdown/types.ts

```diff
@@ -4,12 +4,32 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import {ReactNode, HTMLAttributes, ReactElement, WeakValidationMap, KeyboardEventHandler} from 'react';
+import {
+  RefObject,
+  ReactNode, HTMLAttributes, ReactElement, WeakValidationMap, KeyboardEventHandler} from 'react';
 import * as PropTypes from 'prop-types';
 
 import { NX_BUTTON_VARIANTS, NX_BUTTON_VARIANT_TYPE } from '../NxButton/types';
 import { TooltipConfigProps, tooltipPropTypesShape } from '../../util/tooltipUtils';
 
+export interface AbstractDropdownToggleElementProps {
+  onToggleCollapse: (() => void);
+  toggleRef: RefObject<HTMLButtonElement>;
+}
+
+export type AbstractDropdownToggleElement = (props: AbstractDropdownToggleElementProps) => ReactNode;
```

Comment:
I'd like to draw a bigger distinction between a render function, which is what this is, and a react component proper.  It's worth noting that I think you made the right choice in making this not be an actual component, since it'll typically be a nested function which gets re-instantiated on each render of the parent, which would confuse react and probably degrade performance, as it would think its an entirely new type of component each time.  That said, you've currently got it set up to be an awful lot _like_ a component, which can confuse the reader a little bit.  I think the distinction would be clearer if rather than taking a "props" object, this function simply took he `onToggleCollapse` and `toggleRef` as two separate parameters.

Reply:
> I'd like to draw a bigger distinction between a render function, which is what this is, and a react component proper. It's worth noting that I think you made the right choice in making this not be an actual component, since it'll typically be a nested function which gets re-instantiated on each render of the parent, which would confuse react and probably degrade performance, as it would think its an entirely new type of component each time. That said, you've currently got it set up to be an awful lot _like_ a component, which can confuse the reader a little bit. I think the distinction would be clearer if rather than taking a "props" object, this function simply took the `onToggleCollapse` and `toggleRef` as two separate parameters.

Thanks so much, yeah that makes sense: I updated it here: 50b69374e3165bcd5ab1febc84be2c3716211632